app/models/setting/general.rb

```diff
@@ -29,6 +29,8 @@ def self.default_settings
       self.set('instance_id', N_("Foreman instance ID, uniquely identifies this Foreman instance."), 'uuid', N_('Foreman UUID'), Foreman.uuid),
       self.set('default_locale', N_("Language to use for new users"), nil, N_('Default language'), nil, { :collection => Proc.new { locales } }),
       self.set('default_timezone', N_("Timezone to use for new users"), nil, N_('Default timezone'), nil, { :collection => Proc.new { timezones } }),
+      self.set('enable_facts_storage', N_("Enable Fact Values feature"), FactValue.unscoped.any?, N_('Enable Fact Values feature')),
+      self.set('enable_reports_storage', N_("Enable Reports feature"), ConfigReport.unscoped.any?, N_('Enable Reports feature')),
```

Comment:
This would be probably more appropriate (the same for facts):

```suggestion
      self.set('enable_reports_storage', N_("Store Reports"), ConfigReport.unscoped.any?, N_('Store reports in the database including historical records')),
```